### ISSUE TITLE
Cannot parse EXTENDED_GLOB

### ISSUE THREAD

**Issue by denysdovhan** [koalaman/shellcheck#855]
#### For bugs

- Rule ids: `SC1036`, `SC1056`, `SC1072`, `SC1073`, `SC1009`
- My shellcheck version: `0.4.5` 
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

I have a zsh-theme called [Spaceship](https://github.com/denysdovhan/spaceship-zsh-theme) and [we use `EXTENDED_GLOB` option for zsh](https://github.com/denysdovhan/spaceship-zsh-theme/blob/master/spaceship.zsh#L470-L474). Unfortunately, shellcheck fails when it meet [such expressions](https://github.com/denysdovhan/spaceship-zsh-theme/blob/master/spaceship.zsh#L253-L254):

```sh
# Show Pyenv status only for Python-specific folders
[[ -f requirements.txt || -n *.py(#qN) ]] || return
```

#### Here's what shellcheck currently says:

```sh

In spaceship.zsh line 251:
spaceship_pyenv_status() {
^-- SC1009: The mentioned parser error was in this function.
                         ^-- SC1073: Couldn't parse this brace group.


In spaceship.zsh line 255:
  test -f requirements.txt || test -n *.py(#qN) || return
                                          ^-- SC1036: '(' is invalid here. Did you forget to escape it?
                                          ^-- SC1056: Expected a '}'. If you have one, try a ; or \n in front of it.
                                          ^-- SC1072: Missing '}'. Fix any mentioned problems and try again.

```

#### Here's what I wanted or expected to see:

Parse EXTENDED_GLOB globs as well as others.

**Comment by koalaman** [comment]
Shellcheck supports bash, ksh, dash and sh, but does not support zsh at this time. Sorry.

**Comment by denysdovhan** [comment]
@koalaman do you plan to add zsh support? Should I wait for it?

**Comment by koalaman** [comment]
There are no current plans, but you can follow #809 for any updates on this.
